src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -71,10 +72,10 @@ public void validate() throws MimeTypeParseException {
   @DataBoundSetter
   public void setUri(URL url) throws URISyntaxException
   {
-    this.uri = url.toURI();
+    this.uri = url.toExternalForm();
   }
 
-  public void setUri(URI uri) throws URISyntaxException
+  public void setUri(String uri) throws URISyntaxException
```

Comment:
Can you add a new setter instead? I don't like changing the field type to String, this should not be neccessary

Reply:
actually we have to change to string.
Config as code will fail with 
"Cannot find configurator for type class java.net.URI"

Reply:
I tried locally and it was ok, but next run it failed again. So order is not reliable.

Reply:
You mean it's not enough to have the String setter defined first?

Not supporting URI fields is strange to me. Is there an open issue about it in JCasc or something?

Reply:
Obviously not. I know that URL is supported. But I think casc is inspecting the class and finds it is of type URI and then fails.
The change itself is not binary compatible but the configuration is compatible, so no need to do any migration

Reply:
So what would it take to have JCasc process URI? This seems like a simple change to add support for this type

Reply:
It is possible to add a converter. CASC uses Stapler which exposes a public static field, where one can register a converter.
With that we can stay with URI

Reply:
Nice, I didn't know about it.